notebook-lm-clone/src/podcast/script_generator.py

```
@@ -0,0 +1,299 @@
+            response = self.llm.call(prompt)
+            script_data = json.loads(response)
+            
+            if 'script' not in script_data or not isinstance(script_data['script'], list):
+                raise ValueError("Invalid script format returned by LLM")
+            
+            validated_script = self._validate_and_clean_script(script_data['script'])
+            
+            return {'script': validated_script}
+            
+        except json.JSONDecodeError as e:
+            logger.error(f"Failed to parse LLM response as JSON: {e}")
```

Ruff (0.13.1)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)